### DISCUSSION TITLE
Improve launcher icon setup

### DISCUSSION THREAD

**Discussion by JonatanPlesko** [infinum/android_dbinspector#61] [Ideas]
Currently, you have to create an AndroidManifest.xml in the debug folder with the DbInspectorActivity defined in it to avoid showing the launcher icon in release builds. You also have to duplicate your main activity definition and place it above the DbInspector one - otherwise, the debug manifest takes priority and DbInspectorActivity is considered app's main activity.

In previous versions, this behaviour was all automatic. I think we should either revert to that behaviour or find a way to make the setup much simpler and without requiring creating special manifests in debug folders with duplicated activities.

**Comment by bojankoma** [comment]
I agree with half of this and disagree with the other half. So let's come to some sort of conclusion together.
Creating another launcher activity in your projects setup is not ideal and obviously brings in confusion.
We won't revert back to always automatic approach because it has its own problems and caveats.
I'll try to come up with some solution and put it up here for discussion.

**Reply by JonatanPlesko** [reply]
What would be the problems and caveats of an always automatic approach?

**Reply by bojankoma** [reply]
You have to explicitly remove the node from manifest. 
So basically, you still have to write it inside the manifest.

**Reply by JonatanPlesko** [reply]
So, either way, you have to write the manifest. In that case, the discussion is more semantical than technical. Do we think most users would prefer this behaviour by default or not?

I'm inclined to say _yes_ since it is only for debug builds whose purpose is exactly that - debugging.

Ideally, of course, we go for an alternative solution that doesn't require Manifest wizardry.

**Comment by bojankoma** [comment]
Check out the behaviour of 5.2.0 release.

**Reply by JonatanPlesko** [reply]
It's an improvement for sure! Though the fact that I have to add it to the AndroidManifest.xml anyway to keep the same label as the app name still bothers me a bit. Is it possible to make that as a default? How was it set up before?

**Comment by bojankoma** [comment]
This has been resolved in 5.2.6 to joy of all included stakeholders.
